src/chart/components/pane/pane-manager.component.ts

```diff
@@ -200,6 +205,90 @@ export class PaneManager extends ChartBaseElement {
 		}
 	}
 
+	/**
+	 * Hides a pane from the chart and all related components
+	 */
+	public hidePane(paneUUID: string) {
+		const pane = this.panes[paneUUID];
+		// hide pane only if we have more than one visible pane
+		if (pane !== undefined && pane.getState() !== 'hidden') {
+			const paneId = CanvasElement.PANE_UUID(paneUUID);
+			const paneResizerId = CanvasElement.PANE_UUID_RESIZER(paneUUID);
+			const resizer = this.userInputListenerComponents.find(
+				el => el instanceof BarResizerComponent && el.id === paneResizerId,
+			);
+			resizer?.disable();
+
+			const heightRatio = this.canvasBoundsContainer.graphsHeightRatio;
+			const paneHeightRatio = heightRatio[paneUUID];
+			const ratioToAdd =
+				paneHeightRatio / (Object.values(this.panes).filter(p => p.getState() !== 'hidden').length - 1);
+
+			const newHeightRatio: Record<string, number> = {};
+			for (const key in this.panes) {
+				if (Object.getOwnPropertyDescriptor(this.panes, key)) {
+					if (key === paneUUID || this.panes[key].getState() === 'hidden') {
+						newHeightRatio[key] = 0;
+						continue;
+					}
+					newHeightRatio[key] = heightRatio[key] + ratioToAdd;
+				}
+			}
+
+			pane.hide();
+			this.canvasBoundsContainer.graphsHeightRatio = newHeightRatio;
+			this.canvasBoundsContainer.recalculatePanesHeightRatios();
+			this.canvasBoundsContainer.bounds[paneId].height = 0;
+			this.canvasBoundsContainer.bounds[paneResizerId].height = 0;
+			this.recalculateState();
+			this.canvasBoundsContainer.paneVisibilityChangedSubject.next();
+		}
+
+		return this.canvasBoundsContainer.graphsHeightRatio;
+	}
+
+	/**
+	 * Shows a pane, use if the pane is hidden
+	 */
+	public showPane(paneUUID: string): Record<string, number> {
+		const pane = this.panes[paneUUID];
+		const paneResizerId = CanvasElement.PANE_UUID_RESIZER(paneUUID);
+		if (pane !== undefined && pane.getState() === 'hidden') {
+			const resizer = this.userInputListenerComponents.find(
+				el => el instanceof BarResizerComponent && el.id === paneResizerId,
+			);
+			resizer?.enable();
+			const defaultHeightRatio = getHeightRatios(Object.keys(this.panes).length - 1);
+			const newHeightRatio: Record<string, number> = {};
+			for (const key in this.panes) {
```

Comment:
looks like logic of caclulation new height ratios should be placed in this.canvasBoundsContainer.recalculatePanesHeightRatios(); based on pane's state